Dust off the Python interface

d-torrance opened the pull request:
I've often been curious about the bits of Python-related code in the source, so I figured I'd play around with it and try to get it running with Python 3.

I'm not sure how useful it is, but running `./configure --with-python` works now, and we can do some simple things:

```m2
i1 : importFrom_Core {"runSimpleString"};

i2 : runSimpleString "print('Hello, world!')"
Hello, world!

i3 : runSimpleString "import sys"

i4 : runSimpleString "print(sys.version)"
3.9.5 (default, May 11 2021, 08:20:37) 
[GCC 10.3.0]

i5 : version#"python version"

o5 = 3.9.5
```

mahrud commented:
Can it get [Python.m2](https://github.com/Macaulay2/M2/blob/master/M2/Macaulay2/packages/undistributed-packages/Python.m2) or [Sage.m2](https://github.com/Macaulay2/M2/blob/master/M2/Macaulay2/packages/undistributed-packages/Sage.m2) to work again?

d-torrance commented:
> Can it get [Python.m2](https://github.com/Macaulay2/M2/blob/master/M2/Macaulay2/packages/undistributed-packages/Python.m2) or [Sage.m2](https://github.com/Macaulay2/M2/blob/master/M2/Macaulay2/packages/undistributed-packages/Sage.m2) to work again?

* `Python`: Yes, absolutely, after switching `runString` to `runPythonString` because of 7285468:

```m2
i1 : loadPackage("Python", FileName => "~/src/macaulay2/M2/M2/Macaulay2/packages/undistributed-packages/Python.m2");
-- success: python is present

i2 : math = context "from math import *";

i3 : math "x = sin(3.4);"

i4 : math "sin(3.4)"

o4 = -0.2555411020268312

o4 : PythonObject of class float

i5 : math "x"

o5 = -0.2555411020268312

o5 : PythonObject of class float

i6 : math "e"

o6 = 2.718281828459045

o6 : PythonObject of class float
```

* `Sage` needs some more work:
```m2
i7 : loadPackage("Sage", FileName => "~/src/macaulay2/M2/M2/Macaulay2/packages/undistributed-packages/Sage.m2");
/usr/lib/python3/dist-packages/apport/report.py:13: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import fnmatch, glob, traceback, errno, sys, atexit, locale, imp, stat
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "", line 1, in <module>
  File "/usr/lib/python3/dist-packages/sage/all.py", line 119, in <module>
    from cysignals.signals import (AlarmInterrupt, SignalError,
  File "/usr/lib/python3/dist-packages/cysignals/__init__.py", line 3, in <module>
    init_cysignals()
  File "src/cysignals/signals.pyx", line 251, in cysignals.signals.init_cysignals
  File "/usr/lib/python3.9/signal.py", line 47, in signal
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
ValueError: signal only works in main thread of the main interpreter
Macaulay2/packages/undistributed-packages/Sage.m2:5:6:(2):[7]: error: python error
```

d-torrance commented:
I went ahead and moved `Python.m2` back into the main packages directory, and also added a few new arithmetic methods:

```m2
i1 : needsPackage "Python";
-- success: python is present

i2 : x = rs "5"

o2 = 5

o2 : PythonObject of class int

i3 : y = rs "2"

o3 = 2

o3 : PythonObject of class int

i4 : x + y

o4 = 7

o4 : PythonObject of class int

i5 : x - y

o5 = 3

o5 : PythonObject of class int

i6 : x * y

o6 = 10

o6 : PythonObject of class int

i7 : x / y

o7 = 2.5

o7 : PythonObject of class float
```

d-torrance commented:
I've been playing around with this some more, and it's very cool!  In [another branch](https://github.com/d-torrance/M2/tree/python3-dev), I've written a number of additional D wrappers around Python C functions and have used them to convert back and form between basic Macaulay2 and Python types.

Here's one quick example:

```m2
i1 : loadPackage "Python";
 -- success: python is present

i2 : normalDist = context "from scipy.stats import norm";

i3 : normalProb = toMacaulay2 normalDist "norm.cdf"

o3 = normalProb

o3 : FunctionClosure

i4 : normalProb 1.96

o4 = .97500210485178

o4 : RR (of precision 53)
```
